torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,436 @@
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+        sentence_eed:
+            list of sentence-level scores
+
+    Return:
+        individual sentence scores as a list of Tensors
+    """
+    reference_corpus, hypothesis_corpus = _preprocess_sentences(reference_corpus, hypothesis_corpus, language)
+
+    if sentence_eed is None:
+        sentence_eed = []
+
+    # return tensor(0.0) if reference_corpus or hypothesis_corpus is empty
+    if 0 in (len(hypothesis_corpus), len(reference_corpus[0])):
+        return sentence_eed
+
+    for reference_words, hypothesis in zip(reference_corpus, hypothesis_corpus):
+        score = _compute_sentence_statistics(reference_words, hypothesis, alpha, rho, deletion, insertion)
+        sentence_eed.append(score)
+
+    return sentence_eed
+
+
+def eed(
+    reference_corpus: Sequence[Union[str, Sequence[str]]],
+    hypothesis_corpus: Union[str, Sequence[str]],
+    language: Literal["en", "ja"] = "en",
+    return_sentence_level_score: bool = False,
+    alpha: float = 2.0,
+    rho: float = 0.3,
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+) -> Union[Tensor, Tuple[Tensor, Tensor]]:
+    """Computes extended edit distance score (`EED`_) [1] for strings or list of strings. The metric utilises the
+    Levenshtein distance and extends it by adding an additional jump operation.
+
+    Args:
+        reference_corpus:
+            An iterable of iterables of reference corpus.
+        hypothesis_corpus:
+            An iterable of hypothesis corpus.
+        language:
+            Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+        return_sentence_level_score:
+            An indication of whether sentence-level EED is to be returned.
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+
+    Return:
+        Extended edit distance score as a tensor
+
+    Example:
+        >>> from torchmetrics.functional import eed
+        >>> reference_corpus = ["this is the reference", "here is another one"]
+        >>> hypothesis_corpus = ["this is the prediction", "here is an other sample"]
+        >>> eed(reference_corpus=reference_corpus, hypothesis_corpus=hypothesis_corpus)
+        tensor(0.3204)
+
+    References:
+        [1] P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”,
+        submitted to WMT 2019. `EED`_
+    """
+    # input validation for parameters
+    for parameter in (alpha, rho, deletion, insertion):
+        assert parameter >= 0
+        assert isinstance(parameter, float)
```

Comment:
Please use rather something like:
```python
if isinstance(alpha, float) or alpha < 0:
    raise ValueError("Parameter `alpha` is expected to be a non-negative float.")
```
instead of `assert` statements.

Reply:
got it